Exiled.Events/Patches/Fixes/OverwatchSkipFix.cs

```diff
@@ -0,0 +1,87 @@
+// -----------------------------------------------------------------------
+// <copyright file="OverwatchSkipFix.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Fixes
+{
+#pragma warning disable SA1118
+    using System.Collections.Generic;
+    using System.Reflection;
+    using System.Reflection.Emit;
+
+    using HarmonyLib;
+
+    using NorthwoodLib.Pools;
+
+    using UnityEngine;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Fixes <see cref="CharacterClassManager.RunDefaultClassPicker(bool, out int[], out System.Collections.Generic.Dictionary{UnityEngine.GameObject, RoleType})"/> method.
+    /// </summary>
+    [HarmonyPatch(typeof(CharacterClassManager), nameof(CharacterClassManager.RunDefaultClassPicker))]
+    internal static class OverwatchSkipFix
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            // The index offset
+            int offset = 2;
+            int lastBrOffset = 6;
+
+            // Find the first Fraction.Others call.
+            int index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldc_I4_3) + offset;
+
+            // Find labels.
+            Label continueLabel = (Label)newInstructions[index - 1].operand;
+            Label originalLabel = (Label)newInstructions[index - lastBrOffset].operand;
+
+            // Fix original label.
+            Label fixLabel = generator.DefineLabel();
+            newInstructions[index - lastBrOffset].operand = fixLabel;
+
+            // if (referenceHub.serverRoles.OverwatchEnabled)
+            // {
+            //      list.Add(RoleType.Spectator);
+            //      playersRoleList.Add(referenceHub.gameObject, RoleType.Spectator);
+            //      continue;
+            // }
+            newInstructions.InsertRange(index, new[]
+            {
+                // if (referenceHub.serverRoles.OverwatchEnabled)
+                //      continue;
```

Comment:
These comments seem to contradict each other. I would remove the `//      continue;` from the second one, and add it to where the continue actually is.

Reply:
Oh lmao looks like this is already fixed. Well, that was the only issue I saw.